modules/apm/src/test/java/org/elasticsearch/telemetry/apm/internal/metrics/RecordingMeterProvider.java

```diff
@@ -0,0 +1,645 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.telemetry.apm.internal.metrics;
+
+import io.opentelemetry.api.common.Attributes;
```

Comment:
do we really have to implement an otel api in test framework?
can we not limit ourselfs to just implementing the recording version of the `org.elasticsearch.telemetry.metric.Meter`?